google_gke/variables.tf

```diff
@@ -1,101 +1,242 @@
-variable "region" {
-  default = "us-central1"
-}
-
-variable "environment" {
-  default = ""
+#
+# Cluster Metadata
+#
+variable "description" {
+  default     = null
+  description = "The description of the cluster"
+  type        = string
 }
 
 variable "kubernetes_version" {
-  description = "Version of kubernetes to setup"
   default     = "latest"
+  description = "The Kubernetes version of the masters. If set to 'latest' it will pull latest available version. Defaults to 'latest'."
+  type        = string
 }
 
-variable "release_channel" {
+variable "labels" {
+  default     = {}
+  description = "The GCE resource labels (a map of key/value pairs) to be applied to the cluster & other cluster-related resources. Merged with default labels (see locals.tf)."
+  type        = map(string)
+}
+
+variable "name" {
+  description = "Name of the cluster or application (required)."
+  type        = string
+}
+
+variable "project_id" {
+  default     = null
+  description = "The project ID to host the cluster in."
+  type        = string
+}
+
+variable "realm" {
+  description = "Name of infrastructure realm (e.g. prod or nonprod)."
+  type        = string
+
+  validation {
+    condition     = contains(["mgmt", "nonprod", "prod"], var.realm)
+    error_message = "Valid values for realm: mgmt, nonprod, prod."
+  }
+}
+
+variable "region" {
+  default     = "us-central1"
```

Comment:
Do we want to set this to `null` so it will use the provider's default region?

Reply:
going to try shortly.